linux/LinuxProcessTable.c

```diff
@@ -1098,6 +1098,41 @@ static void LinuxProcessTable_readCwd(LinuxProcess* process, openat_arg_t procFd
    free_and_xStrdup(&process->super.procCwd, pathBuffer);
 }
 
+/*
+ * Read /proc/<pid>/syscall (thread-specific data)
+ */
+#ifdef HAVE_SYSCALL
+static void LinuxProcessTable_readSyscall(LinuxProcess* process, openat_arg_t procFd) {
+   char buffer[1024];
+
+   ssize_t r = xReadfileat(procFd, "syscall", buffer, sizeof(buffer));
+   if (r <= 0) {
+      goto failed;
+   }
+
+   char* numPtr = buffer;
+   process->syscall_num = fast_strtoull_dec(&numPtr, r);
+   if (*numPtr == ' ') {
+      process->syscall_state = SYSCALL_STATE_CALLING;
+      Row_updateFieldWidth(SYSCALL, snprintf(NULL, 0, "%d", process->syscall_num));
+      return;
+   }
+
+   char* eol = String_strchrnul(buffer, '\n');
+   *eol = '\0';
+
+   if (String_eq(buffer, "running")) {
```

Comment:
You can use `String_startsWith(buffer, "running")` and free the need of assigning `*eol = '\0'` on the line above.